### ISSUE TITLE
Minecraft server process failed {"exitCode": -1}

### ISSUE THREAD

**Issue by k0mar12** [itzg/docker-minecraft-server#1665]
### Describe the problem

Runing on the:
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

### Container definition

```
version: '3'

services:
  mc-server:
    image: itzg/minecraft-server
    user: minecraft
    container_name: '${DOCKER_PREFIX}_mc_server'
    ports:
      - 25565:25565
      - 25575:25575
    tty: true
    stdin_open: true
    restart: unless-stopped
    environment:
      EULA: 'TRUE'
      TYPE: 'PAPER'
      VERSION: '1.19.2'
      UID: 1000
      GID: 998
      ONLINE_MODE: 'FALSE'
      MOTD: '${MC_SERVER_NAME}'
      MEMORY: '${MC_RAM_MEMORY}'
      MAX_PLAYERS: '${MC_MAX_PLAYERS}'
    volumes:
      - ./src/minecraft-server/plugins:/data/plugins
      - ./src/minecraft-server/mods:/data/mods
      - ./src/minecraft-server/config:/data/config

id`s local user is 1000:998
```

### Container logs

```
[18:40:13 INFO]: Running delayed init tasks
[18:40:13 INFO]: Done (30.105s)! For help, type "help"
[18:40:13 INFO]: Timings Reset
2022-08-11T18:40:23.192Z	WARN	mc-server-runner	Minecraft server process failed	{"exitCode": -1}
2022-08-11T18:40:23.196Z	INFO	mc-server-runner	Done
```

**Comment by itzg** [comment]
That's the output after the container startup is finished. So it's the Minecraft server software itself failing. 

One thing to investigate is add `DEBUG_MEMORY: "true"` to the environment and provide the **full** container logs, mainly the `[init]` lines.

**Comment by k0mar12** [comment]
@itzg 
```
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 5 1000 1000 4096 Aug 12 00:01 /data'
[init] Resolved version given 1.19.2 into 1.19.2 and major version 1.19
[init] Resolving type given PAPER
[init] Removing old PaperMC versions ...
[init] Downloading PaperMC 1.19.2 (build 125) ...
[init] Creating server.properties in /data/server.properties
[init] Disabling whitelist functionality
[init] Setting white-list to 'false' in /data/server.properties
[init] WARNING: whitelist enabled but not enforced. Set ENFORCE_WHITELIST=TRUE or update 'enforce-whitelist' in server.properties to enforce the whitelist.
[init] Setting enable-rcon to 'true' in /data/server.properties
[init] Setting rcon.password to 'minecraft' in /data/server.properties
[init] Setting rcon.port to '25575' in /data/server.properties
[init] Setting max-players to '5' in /data/server.properties
[init] Setting online-mode to 'false' in /data/server.properties
[init] Setting motd to 'Pokess' in /data/server.properties
[init] Checking for JSON files.
[init] Memory usage and availability (in MB)
Linux 2edf47cc0001 5.4.0-39-generic #43-Ubuntu SMP Fri Jun 19 10:28:31 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
              total        used        free      shared  buff/cache   available
Mem:           1971         699         766           3         506        1111
Swap:             0           0           0
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
Downloading mojang_1.19.2.jar
Applying patches
Starting org.bukkit.craftbukkit.Main
System Info: Java 17 (OpenJDK 64-Bit Server VM 17.0.4+8) Host: Linux 5.4.0-39-generic (amd64)
Loading libraries, please wait...
2022-08-12 00:01:46,596 ServerMain WARN Advanced terminal features are not available in this environment
[00:01:51 INFO]: Building unoptimized datafixer
[00:01:53 INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[00:01:53 INFO]: Found new data pack file/bukkit, loading it automatically
[00:01:56 INFO]: Loaded 7 recipes
[00:01:58 INFO]: Starting minecraft server version 1.19.2
[00:01:58 INFO]: Loading properties
[00:01:58 INFO]: This server is running Paper version git-Paper-125 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 3f0415b)
[00:01:58 INFO]: Server Ping Player Sample Count: 12
[00:01:58 INFO]: Using 4 threads for Netty based IO
[00:01:58 INFO]: Default game type: SURVIVAL
[00:01:58 INFO]: Generating keypair
[00:01:59 INFO]: Starting Minecraft server on *:25565
[00:01:59 INFO]: Using epoll channel type
[00:01:59 INFO]: Paper: Using libdeflate (Linux x86_64) compression from Velocity.
[00:01:59 INFO]: Paper: Using OpenSSL 1.1.x (Linux x86_64) cipher from Velocity.
[00:02:00 INFO]: [LuckPerms] Loading LuckPerms v5.4.41
[00:02:00 INFO]: [Vault] Loading Vault v1.7.3-b131
[00:02:00 INFO]: [LuckPerms] Enabling LuckPerms v5.4.41
[00:02:01 INFO]:         __    
[00:02:01 INFO]:   |    |__)   LuckPerms v5.4.41
[00:02:01 INFO]:   |___ |      Running on Bukkit - Paper
[00:02:01 INFO]: 
[00:02:01 INFO]: [LuckPerms] Loading configuration...
[00:02:01 INFO]: [LuckPerms] Loading storage provider... [H2]
[00:02:02 INFO]: [LuckPerms] Loading internal permission managers...
[00:02:02 INFO]: [LuckPerms] Performing initial data load...
[00:02:02 INFO]: [LuckPerms] Successfully enabled. (took 2342ms)
[00:02:02 INFO]: [Vault] Enabling Vault v1.7.3-b131
[00:02:03 INFO]: [Vault] [Permission] SuperPermissions loaded as backup permission system.
[00:02:03 INFO]: [Vault] Enabled Version 1.7.3-b131
[00:02:03 INFO]: [LuckPerms] Registered Vault permission & chat hook.
[00:02:03 WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[00:02:03 WARN]: The server will make no attempt to authenticate usernames. Beware.
[00:02:03 WARN]: While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
[00:02:03 WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[00:02:03 INFO]: Preparing level "world"
[00:02:03 WARN]: Failed to parse level-type default, defaulting to minecraft:normal
[00:02:31 INFO]: Preparing start region for dimension minecraft:overworld
[00:02:32 INFO]: Preparing spawn area: 0%
[00:02:32 INFO]: Preparing spawn area: 0%
[00:02:32 INFO]: Preparing spawn area: 0%
[00:02:32 INFO]: Time elapsed: 1144 ms
[00:02:32 INFO]: Preparing start region for dimension minecraft:the_nether
[00:02:33 INFO]: Time elapsed: 781 ms
[00:02:33 INFO]: Preparing start region for dimension minecraft:the_end
[00:02:34 INFO]: Time elapsed: 1113 ms
[00:02:34 INFO]: Starting remote control listener
[00:02:34 INFO]: Thread RCON Listener started
[00:02:34 INFO]: RCON running on 0.0.0.0:25575
[00:02:34 INFO]: Running delayed init tasks
[00:02:34 INFO]: Done (36.476s)! For help, type "help"
[00:02:34 INFO]: Timings Reset
[00:02:34 INFO]: [Vault] Checking for Updates ... 
[00:02:34 INFO]: [Vault] No new version available
[00:03:19 WARN]: Can't keep up! Is the server overloaded? Running 5052ms or 101 ticks behind
```

Aha, probably your right :) I think, i have not enough free ram memory, am i right ?)

**Comment by k0mar12** [comment]
@itzg Yes it is, thanks. Problem in my low memory value.
